### ISSUE TITLE
[Bug Report][3.3.22] V-Dialog close-on-back does not work

### ISSUE THREAD

**Issue by madnik7** [vuetifyjs/vuetify#18505]
### Environment
**Vuetify Version:** 3.3.22
**Vue Version:** 3.3.4
**Browsers:** Chrome 118.0.0.0
**OS:** Windows 10

### Steps to reproduce
close-on-back does nothing

### Expected Behavior
It should close the dialog by pressing the browser back button. Crucial for mobile apps,

### Actual Behavior
Does nothing

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNp9U8FOIzEM/RUrZ2aKYPeCymrR7hGJD6Ac3MQUi0wSJs6AVPHveKZMmU4LpyT2e/ZzbN9vTW7t4ialuitkrsxSqEkehf6sAsCyqzCl4To8bAyCHKj9NA1Gx+jjBl7ZydP1yvw+P18ZsD5mqmKo1mif92jFj/Ghq7KPcoVWuEOJrVK3kNqYMryvzIQyJFlLUMaag1PcgNIkQm+iz7tEAf4PKpQIy8UAnyZdTKr6QYujRyy+D7kFzje9MjolxmLrQFg8KXSfeAraw6pe48wFcBtbaoBTLg246GMLmQWwITkD/eNMVkhKC+g4cbYcNkCe1ZnJKWEejrjkJqokrUdjcbDs2JUgUAQ8rjUbkOwyETS4CQjo+aVgPROtXzdVfbqkvmUq8qgq9eeEVqejj/N5nQcZuzm3wtjNf/3k7Nt5DPtrPdtnBY4dqjv0heAaHtFnOqYMag4H4qDWU+WMzu+G6Au1m/5xRXrawY70lt0KHfDN+5m5rC/rX2Z3XlyYhw+Iowqe)

**Comment by websitevirtuoso** [comment]
No it doesn't. You can open modal using route. and in this case when click back in browser it will just close your modal window

**Comment by paul-thebaud** [comment]
Hello,
Any news on this please? I'm facing the same issue.

**Comment by paul-thebaud** [comment]
I investigated this. Looks like the property only work when the dialog is a child (deep or not) of the `router-view` component.

**Comment by paul-thebaud** [comment]
@hesam64436, I got a lot of dialogs in my app. Dialogs which are child of a route component (defined in router routes) have a working "close on back", but dialogs outside of the app routing (such as in my app's drawer) does not work.

**Comment by paul-thebaud** [comment]
@hesam64436 here it is (not tested, just here for the example).

## router.ts

```ts
const routes = [
  { path: '/', component: HomeView },
];

export default createRouter({
  history: createWebHistory(),
  routes,
});
```

## App.vue

```vue
<template>
  <v-app>
    <!-- KO -->
    <v-dialog close-on-back />
    <v-container>
      <router-view />
    </v-container>
  </v-app>
</template>
```

## HomeView.vue

```vue
<template>
  <div>
    <!-- OK -->
    <v-dialog close-on-back />
  </div>
</template>
```

**Comment by MetRonnie** [comment]
I've found that a dialog with `close-on-back` only closes on forward navigation, not back

**Comment by J-Sek** [comment]
> I wrote my own handlers on nuxt2 and moved them into nuxt4 but today saw this option on v-dialog docs. used it and it worked. but now that i was cleaning the code it stop working so i return back to my own logic!! (-_-)

Yeah, I am playing with it in Nuxt3 app and it needs a bunch. I am still figuring out whether to show confirm/alert or not. Leaving the snippet below for anyone interested to hopefully save half an hour on figuring out the workaround for `close-on-back`.

```ts
import { useEventListener } from '@vueuse/core'

useEventListener(window, 'popstate', (e: Event) => {
  if (dialogFormOpen.value) {
    dialogFormOpen.value = false
    e.preventDefault()
  }
})

useEventListener(window, 'beforeunload', (e: Event) => {
  console.log('beforeunload')
  if (dialogFormOpen.value) {
    e.preventDefault()
    // const result = confirm('There are some unsaved changes. Discard?')
    // if (!result) e.preventDefault()
  }
})

onBeforeRouteLeave((_to, _from, next) => {
  console.log('onBeforeRouteLeave')
  if (dialogFormOpen.value) {
    next(false)
    // const result = confirm('There are some unsaved changes. Discard?')
    // next(result)
    return
  }
  next()
})
```

**Comment by qwert21ant** [comment]
Hey, I'm facing the same issue, this `close-on-back` feature doesn't work when current web history entry (vue-router defines it [here](https://github.com/vuejs/router/blob/main/packages/router/src/history/html5.ts)) has `position = 0`, or previous entry (with position x-1) has `replaced = true` (and this is at least first element of history, that appears when opening fresh browser tab and then navigating to some url).
It's happening because of this check for `e.state?.replaced`: https://github.com/vuetifyjs/vuetify/blob/3b0236d4520ba683ccb8e5026c90ec5d1fd05dfc/packages/vuetify/src/composables/router.tsx#L143-L148
Is this check really necessary?

**Comment by johnleider** [comment]
@qwert21ant's diagnosis above is correct and still reproduces on **Vuetify 4.2.0** with both **vue-router 4.6.4 and 5.1.0**. Independently confirmed today by @flanaganvaquero in Discord, who ran the elimination. Adding a minimal repro, plus **a second defect that will surface the moment the `replaced` check is fixed**.

### Minimal repro, and the test that distinguishes it

- `/` → `/test`, open dialog on `/test`, press Back → **navigates, dialog stays open**
- `/` → `/a` → `/b`, open dialog on `/b`, press Back → **dialog closes, stays on `/a`** ✅

The difference is only which entry you return *to*. vue-router stamps the initial entry `replaced: true` in `useHistoryStateNavigation`:

```js
if (!historyState.value) changeLocation(currentLocation.value, {
  back: null, current: currentLocation.value, forward: null,
  position: history.length - 1,
  replaced: true,
  scroll: null,
}, true)
```

while `push()` builds new entries through `buildState(...)`, where `replaced` defaults to `false`. popstate delivers the state of the entry being restored, so **going back to the landing page always trips `if (e.state?.replaced) return`** and `popped` is never set. Any app whose dialog sits one navigation deep from entry hits this 100% of the time, which is most reports on this issue.

On the open question — *is the check necessary?* — it only ever guards against treating a `router.replace()` as a back navigation. Keying off `position` (or comparing against the previous entry's position) separates the two cases without swallowing the initial entry.

### Second defect: `inTransition` latches permanently

Independent of the above, and it will bite immediately after any fix to `replaced`:

```ts
let inTransition = false                     // module-level, shared by every overlay

removeBefore = router.beforeEach(() => {
  if (!inTransition) {
    inTransition = true
    return new Promise(resolve => {
      setTimeout(() => resolve(popped ? cb() : undefined))   // async path — works
    })
  }
  return popped ? cb() : undefined                           // sync path — see below
})
removeAfter = router.afterEach(() => { inTransition = false })
```

`inTransition` is cleared **only** in `afterEach`, and Vue Router does not run `afterEach` when a guard aborts a navigation. Closing the dialog *is* an abort (`cb()` returns `false`), so the flag latches `true` after the first successful close and is never cleared for the lifetime of the page.

Once latched, every later back press takes the synchronous branch. `beforeEach` runs from Router's own popstate listener, registered at history creation, whereas Vuetify's `onPopstate` is registered in a `nextTick` after the overlay mounts — so Router's handler runs first and the sync branch reads `popped === false`. The `setTimeout` in the async branch is the only thing that lets `popped` arrive in time.

Net effect after fixing `replaced` alone: **close-on-back works exactly once per page load, then never again**, and because the flag is module-global the first close anywhere poisons every overlay in the app. Any other guard that aborts a navigation poisons it too.

### Why this survived since 2023

`closeOnBack` has no test that can fail on either defect. The only spec touching it (`VDialog.spec.browser.tsx`, `describe('routing back')`):

```js
async function simulateBackNavigation (router) {
  router.back()
  for (let i = 0; i < 10; i++) await nextTick()
  window.dispatchEvent(new PopStateEvent('popstate', { state: {} }))
  await wait()
}
```

- runs on `createMemoryHistory()`, which has **no** popstate listener, so the real listener-ordering behaviour is absent
- dispatches `state: {}`, which has **no `replaced` field**, so the early return is never exercised
- calls `router.back()` *before* dispatching popstate, the reverse of browser order, manufacturing precisely the interleaving that makes the current code pass

A regression test needs `createWebHistory` in a browser context, a real Back, and two cases: returning to the initial entry, and a second consecutive close on the same page load.
